src/irmin-pack/unix/control_file.ml

```diff
@@ -61,12 +73,35 @@ module Make (Io : Io.S) = struct
 
   type t = { io : Io.t; mutable payload : Latest_payload.t }
 
+  let upgrade_v3_to_v4 (pl : Payload_v3.t) : Payload_v4.t =
+    let status =
+      match pl.status with
+      | From_v1_v2_post_upgrade x -> Payload_v4.From_v1_v2_post_upgrade x
+      | From_v3_no_gc_yet -> No_gc_yet
+      | From_v3_used_non_minimal_indexing_strategy ->
+          Used_non_minimal_indexing_strategy
+      | From_v3_gced x ->
+          Gced
+            {
+              suffix_start_offset = x.suffix_start_offset;
+              generation = x.generation;
+              oldest_live_commit_offset = x.suffix_start_offset;
+            }
+      | _ -> assert false
```

Comment:
This is typically the case where we want to exhaustively list the remaining tags instead of using a wildcard for the day we will change the definition. However, it is really not expected that we will touch the V3 definition again, so it's OK to leave it that way.

Reply:
Indeed, not sure why I didn't, thanks